src/hotspot/share/opto/macro.cpp

```diff
@@ -1208,7 +1208,7 @@ Node* PhaseMacroExpand::make_load(Node* ctl, Node* mem, Node* base, int offset,
 
 
 Node* PhaseMacroExpand::make_store(Node* ctl, Node* mem, Node* base, int offset, Node* value, BasicType bt) {
-  Node* adr = basic_plus_adr(base, offset);
+  Node* adr = basic_plus_adr(top(), base, offset);
```

Comment:
Doesn't this cause an assert if make_load or make_store is used with a heap oop?  Isn't that a problem for code like PhaseMacroExpand::initialize_object() that calls make_store() with an object?

Reply:
Yes, I like make_raw_load/make_raw_store or even make_load_raw/make_store_raw.